### ISSUE TITLE
[BUG] Any files added to an archive after >4GB of data has already been added will be corrupted.

### ISSUE THREAD

**Issue by zodopoz** [Touffy/client-zip#33]
**Describe the bug**
Any files added to an archive after >4GB of file has already been added will be corrupted. A single file zipped that is greater than >4GB will not be corrupted, however.

**To Reproduce**
1. Go to 'https://touffy.me/client-zip/demo/worker'
2. Add a large file (I tested with a 6.5GB file), and another small file.
3. Download the zip
4. Run `unzip` CLI tool on Zip (also fails with Archive.app)

**Expected behavior**
Both files should be extracted from the zip successfully

**Actual behavior**
The first file is able to be extracted, but the next file fails:
```
~/Downloads λ unzip demozip.zip 
Archive:  demozip.zip
replace Archive.zip? [y]es, [n]o, [A]ll, [N]one, [r]ename: A
 extracting: Archive.zip             
file #2:  bad zipfile offset (local header sig):  1051
```

**Desktop (please complete the following information):**
 - OS: MacOS Monterey
 - Browser: Chrome
 - Version: 99.0.4844.84

Thank you for this library!

**Comment by Touffy** [comment]
Reproduced. Strangely, the offset reported in this error for the second file equals the file length.

**Comment by Touffy** [comment]
Found a seemingly related bug [there](https://github.com/aonez/Keka/issues/423), and I think I see what's going on.

When client-zip decides that a file needs the Zip64 extra field, it always writes the same 3 attributes in it : the file sizes, and the offset.

However, in our case the second file is small, so its sizes do not overflow 32 bits. The Zip64 attributes should only cover any value that was maxed out (0xffffffff or equivalent) in the basic header. The zip extractor knows that and sees that only the offset was maxed out, so it expects the first attribute in the header to be the offset, not the size.

I'll add some logic to skip the sizes for small files and see it it fixes the bug.

**Comment by Touffy** [comment]
Thanks you for the bug report. The version I just published (2.0.1) seems to fix it. Can you confirm ?

**Comment by zodopoz** [comment]
Yep, seems to work now! 😄
